### ISSUE TITLE
`RombergIntegrator` is not thread-safe

### ISSUE THREAD

**Issue by p-himik** [generateme/fastmath#23]
A top-level form in `fastmath.random`:
```clojure
(defonce ^:private ^RombergIntegrator romberg-integrator (RombergIntegrator.))
```

It makes `integrate-pdf` not thread-safe. I have spent quite a few hours hunting down spurious `TooManyEvaluationsException` only to realize that [when I add more threads in order to try and circumvent the flakiness](https://clojurians.slack.com/archives/C03S1KBA2/p1676926759317809?thread_ts=1676924932.649369&cid=C03S1KBA2), I actually get guaranteed failures.

**Comment by genmeblog** [comment]
Thanks and sorry you hit this. 
It's fixed already in the latest SNAPSHOT (which you can safely use). 

I moved the creation of the integrator for every instance of pdf calculation.

https://github.com/generateme/fastmath/blob/master/src/fastmath/random.clj#L1458

**Comment by p-himik** [comment]
Ah, shoot - I should've checked the `master`. :) Thanks!

**Comment by genmeblog** [comment]
Solved in 2.2.0
